book/scheduling-and-threading.md

```diff
@@ -237,10 +267,33 @@ class Tab:
             self.display_list = []
             self.document.paint(self.display_list)
             self.needs_pipeline_update = False
+
+class Browser:
+    def raster_and_draw(self):
+        self.raster_chrome()
+        self.raster_tab()
+        self.draw()
 ```
 
-Replace all cases where parts of the rendering pipeline were called with
-`set_needs_pipeline_update`, for example `load`:
+The last piece is to actually call `run_animation_frame` from somewhere; at the
+moment all we're doing is setting `run_pipeline_now`, which is not too useful
+on its own! This needs to happen in the main event loop, just as I diagrammed
+in the previous section:
+
+``` {.python expected=False}
+    while True:
+        if sdl2.SDL_PollEvent(ctypes.byref(event)) != 0:
+            # ...
+        active_tab = browser.tabs[browser.active_tab]
+        if active_tab.run_pipeline_now:
+            active_tab.run_pipeline_now = False
+            active_tab.run_animation_frame()
+```
```

Comment:
Is here the right place to explain why we must set `run_pipeline_now` before running the animation frame? It is related somehow to the idea that you need to be able to schedule a rAF inside a rAF callback maybe...